test/mongoConnectionStrings.test.ts

```diff
@@ -5,55 +5,83 @@
 
 // The module 'assert' provides assertion methods from node
 import * as assert from 'assert';
-import { getDatabaseNameFromConnectionString } from '../src/mongo/mongoConnectionStrings';
+import { getDatabaseNameFromConnectionString, addDatabaseToAccountConnectionString } from '../src/mongo/mongoConnectionStrings';
 
-function testDatabaseFromConnectionString(connectionString: string, expectedDatabase: string | undefined) {
-    let database = getDatabaseNameFromConnectionString(connectionString);
-    assert.equal(database, expectedDatabase);
+function testDatabaseToAccountConnectionString(connectionString: string, databaseName: string, expectedConnectionString: string | undefined) {
+    let databaseConnectionString = addDatabaseToAccountConnectionString(connectionString, databaseName);
+    assert.equal(databaseConnectionString, expectedConnectionString);
+}
+
+function testDatabaseNameFromConectionString(connectionString: string, expectedDatabaseName: string | undefined) {
+    let databaseName = getDatabaseNameFromConnectionString(connectionString);
+    assert.equal(databaseName, expectedDatabaseName);
+    if (databaseName) {
+        testDatabaseToAccountConnectionString(connectionString, databaseName, connectionString);
+    }
 }
 
 suite(`mongoCollectionStrings`, () => {
     test(`getDatabaseNameFromConnectionString`, () => {
         // Connection strings follow the following format (https://docs.mongodb.com/manual/reference/connection-string/):
         // mongodb://[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]][/[database][?options]]
 
-        testDatabaseFromConnectionString(`mongodb://my-mongo:ayO83FFfUoHE97Jm7WbfnpNCqiF0Yq0za2YmvuLAKYJKf7h7hQaRKWfZfsv8Ux41H66Gls7lVPEKlKm0ueSozg==@your-mongo.documents.azure.com:10255/?ssl=true&replicaSet=globaldb`, undefined);
-        testDatabaseFromConnectionString(`mongodb://my-mongo:ayO83FFfUoHE97Jm7WbfnpNCqiF0Yq0za2YmvuLAKYJKf7h7hQaRKWfZfsv8Ux41H66Gls7lVPEKlKm0ueSozg==@your-mongo.documents.azure.com:10255/our-mongo?ssl=true&replicaSet=globaldb`, "our-mongo");
-        testDatabaseFromConnectionString(`mongodb://my-mongo:ayO83FFfUoHE97Jm7WbfnpNCqiF0Yq0za2YmvuLAKYJKf7h7hQaRKWfZfsv8Ux41H66Gls7lVPEKlKm0ueSozg==@your-mongo.documents.azure.com:10255/?ssl=true&replicaSet=globaldb`, undefined);
+        testDatabaseNameFromConectionString(`mongodb://my-mongo:ayO83FFfUoHE97Jm7WbfnpNCqiF0Yq0za2YmvuLAKYJKf7h7hQaRKWfZfsv8Ux41H66Gls7lVPEKlKm0ueSozg==@your-mongo.documents.azure.com:10255/?ssl=true&replicaSet=globaldb`, undefined);
+        testDatabaseNameFromConectionString(`mongodb://a/b?c`, "b");
```

Comment:
Why'd you change this connection string?

Reply:
Had issues with regex, so changed that for simplifying my fixing process and forgot to turn it back